packages/components/modules/profiles/ProfilePopover/AddProfileModal/index.tsx

```diff
@@ -0,0 +1,179 @@
+
+        if (!mutationErrors?.length) {
+          handleClose()
+        }
+      },
+      onError: console.error,
+    })
+  }
+
+  return (
+    <Dialog
+      open={open}
+      onClose={handleClose}
+      aria-labelledby="organization-modal-title"
+      aria-describedby="organization-modal-description"
+      PaperProps={{
+        sx: {
+          borderRadius: { xs: 0, sm: 2 },
+          minWidth: { xs: '100%', sm: 0 },
+          margin: { xs: 0, sm: 2 },
+          maxWidth: { sm: 366 },
+          height: { xs: '100%', sm: 'auto' },
+          minHeight: { xs: '100%', sm: 'auto' },
+        },
+      }}
+    >
+      <DialogTitle id="organization-modal-title">
+        <Box
+          sx={{ display: 'flex', alignItems: 'center', gap: 2, justifyContent: 'space-between' }}
+        >
+          <Typography variant="h6">{addNewProfileLabel}</Typography>
+          <IconButton onClick={handleClose}>
+            <CloseIcon />
+          </IconButton>
+        </Box>
+      </DialogTitle>
+      <Form onSubmit={form.handleSubmit(onSubmit)}>
+        <DialogContent
+          sx={{ display: 'flex', flexDirection: 'column', gap: 4 }}
+          id="organization-modal-description"
+        >
+          <Typography color="text.secondary">{addNewProfileDescription}</Typography>
+          <Box display="flex" flexDirection="column" gap={2}>
+            <TextField
+              label="Name"
+              name="name"
+              size="medium"
+              control={form.control}
+              onChange={() => {
+                form.trigger('name')
+              }}
+              onBlur={() => {
+                if (!form.getValues('urlPath')) {
+                  form.setValue('urlPath', slugify(form.getValues('name').toLocaleLowerCase()))
+                  form.trigger('urlPath')
+                }
+              }}
+            />
+            <TextField
+              label="URL Path"
+              name="urlPath"
+              size="medium"
+              control={form.control}
+              onChange={() => {
+                form.trigger('urlPath')
+              }}
+            />
+            <Typography color="text.primary">
+              Upon confirming, you agree to our{' '}
+              <Link display="inline" href={termsAndConditionsUrl} target="_blank">
+                Terms and Conditions.
+              </Link>
+            </Typography>
+          </Box>
+        </DialogContent>
+        <DialogActions sx={{ justifyContent: 'flex-end' }}>
+          <Button variant="outlined" color="inherit" onClick={handleClose} sx={{ width: 'auto' }}>
+            Cancel
+          </Button>
+          <LoadingButton
+            variant="contained"
+            color="inherit"
+            sx={{ width: 'auto' }}
+            type="submit"
+            loading={isMutationInFlight}
+            disabled={
+              isMutationInFlight ||
+              Object.keys(form.formState.errors).length > 0 ||
```

Comment:
would it be a case where if `formState.isValid` is true there would still be `formState.errors`? asking because we normally only use isValid